src/passes/reorder_funs.rs

```diff
@@ -0,0 +1,78 @@
+use std::collections::BTreeMap;
+
+use crate::{
+    entity::EntityRef, ExportKind, Func, FuncDecl, FunctionBody, ImportKind, Module, Operator,
+    Terminator, ValueDef,
+};
+
+pub fn reorder_funcs_in_body(b: &mut FunctionBody, f: &BTreeMap<Func, Func>) {
```

Comment:
Could you describe the goal of this pass? To me it reads like a possibly useful utility, but not a core optimization pass that a general compiler backend would have -- would it perhaps work just as well to host it in your use-case directly?